Botnet rebase

sigvartmh opened the pull request:
Lets go haaaam

hakonfam committed:
dump WITH PCD

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
dump

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
dump

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
do we want this?

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

sigvartmh committed:
nrf5340_ netboot: Change how PCD_CMD address work and netboot sequence

Changed how `PCD_CMD_ADDRESS` is calculated. Removed unused buffer in
structure. Changed a bit how netboot boot sequence works.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

sigvartmh committed:
west manifest

sigvartmh committed:
Change cmd address

sigvartmh committed:
Add RAM protection for copy instructionsw

sigvartmh commented on samples/nrf5340/netboot/src/main.c:
```diff
@@ -0,0 +1,66 @@
+		err = pcd_transfer(cmd, rsp, fdev);
+		if (err != 0) {
+			printk("Failed to transfer image: %d. \n\r", err);
+			return;
```
Missing goto. Should we use `while` + `break` instead?

sigvartmh commented on samples/nrf5340/netboot/src/main.c:
```diff
@@ -0,0 +1,66 @@
+	}
+
+	bl_boot(fw_info_find(s0_addr));
+
```
maybe add early return? Since we don't really should set the fail magic

hakonfam committed:
fix net core hex dep (#30)

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
fix dependency issue where net dfu was not generated (#31)

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam commented on cmake/mcuboot.cmake:
```diff
@@ -181,6 +181,34 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
     )
 
+  if (CONFIG_BT_RPMSG_NRF53 AND CONFIG_SOC_NRF5340_CPUAPP)
+    # The bootloader on the network core is enabled. The validation of this
```
Is it? IT is the inclusion of mcuboot on the app core than implies that the net core bootloader is included. Fix this.

hakonfam commented on cmake/mcuboot.cmake:
```diff
@@ -181,6 +181,34 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
+  if (CONFIG_BT_RPMSG_NRF53 AND CONFIG_SOC_NRF5340_CPUAPP)
+    # The bootloader on the network core is enabled. The validation of this
+    # bootloader is performed by MCUBoot on the application core. Hence we
+    # need a target for creating the signed binary of the network core
```
We don't need a target, we need to create the update.

hakonfam commented on cmake/mcuboot.cmake:
```diff
@@ -181,6 +181,34 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
+    # need a target for creating the signed binary of the network core
+    # application.
+
+    include(${CMAKE_BINARY_DIR}/hci_rpmsg/shared_vars.cmake)
```
Add a comment on why this is needed

hakonfam commented on cmake/mcuboot.cmake:
```diff
@@ -181,6 +181,34 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
+
+    include(${CMAKE_BINARY_DIR}/hci_rpmsg/shared_vars.cmake)
+
+    # TODO replace with proper domain once PR is in
```
https://github.com/nrfconnect/sdk-nrf/pull/2527

hakonfam commented on cmake/partition_manager.cmake:
```diff
@@ -358,6 +358,9 @@ if (is_dynamic_partition_in_domain)
   share("set(${domain}_PM_DOMAIN_IMAGES ${prefixed_images})")
   share("set(${domain}_PM_HEX_FILE ${PROJECT_BINARY_DIR}/${merged}.hex)")
+  share("set(${domain}_PM_DOTCONF_FILES ${pm_out_dotconf_files})")
+  # The partition manager script ensures that the 'app' hex always exists.
```
Remove this comment, it gives no info

hakonfam commented on cmake/partition_manager.cmake:
```diff
@@ -382,6 +385,51 @@ else()
+          )
+      endforeach()
+
+      # TODO don't check BOARD, don'c check for BT_RPMSG. Do something smarter, look at domains?
```
https://github.com/nrfconnect/sdk-nrf/pull/2527

Also, this is no longer a BOARD check, but SoC, which is OK

hakonfam commented on include/dfu/pcd.h:
```diff
@@ -0,0 +1,90 @@
+
+/** Magic value written to indicate that a copy should take place. */
+#define PCD_CMD_MAGIC_COPY 0xb5b4b3b6
+/** Magic value written to indicate that a something failed. */
```
```suggestion
/** Magic value written to indicate that something failed. */
```

hakonfam commented on include/dfu/pcd.h:
```diff
@@ -0,0 +1,90 @@
+	void * src_addr;  /* Source address to copy from */
+	size_t len;       /* Number of bytes to copy */
+	size_t offset;    /* Offset to store the flash image in */
+} __attribute__ ((aligned(4)));
```
Is this how alignment is done elsewhere in NCS?

hakonfam commented on include/dfu/pcd.h:
```diff
@@ -0,0 +1,90 @@
+#if !defined(CONFIG_NRF_SPU_RAM_REGION_SIZE) //&& defined nrf53?
+#define CONFIG_NRF_SPU_RAM_REGION_SIZE 0x2000
+#endif
+#define PCD_CMD_ADDRESS (APP_CORE_SRAM_BASE_ADDRESS + APP_CORE_SRAM_SIZE \
```
Add some comments here, why does this need to be at this address? What is it used for?

hakonfam commented on include/dfu/pcd.h:
```diff
@@ -0,0 +1,90 @@
+
+#define APP_CORE_SRAM_BASE_ADDRESS 0x20000000
+#define APP_CORE_SRAM_SIZE KB(512)
+#if !defined(CONFIG_NRF_SPU_RAM_REGION_SIZE) //&& defined nrf53?
```
When is this the case?

spm.c use 

```
/* SPU RAM regions */
#define RAM_SECURE_ATTRIBUTION_REGION_SIZE CONFIG_NRF_SPU_RAM_REGION_SIZE
#define NUM_RAM_SECURE_ATTRIBUTION_REGIONS (TOTAL_RAM_SIZE \
				/ RAM_SECURE_ATTRIBUTION_REGION_SIZE)
```

hakonfam commented on include/dfu/pcd.h:
```diff
@@ -0,0 +1,90 @@
+ *  The command is used to communicate information between the sender
+ *  and the receiver of the DFU image.
+ */
+struct pcd_cmd {
```
Could we move this inside the  C file, and only have an empty struct definition here? To protect the members, and then update the signature of the `pcd_transfer` method to instead take src, len and offset as arguments.

hakonfam commented on samples/bluetooth/peripheral_uart/prj.conf:
```diff
@@ -51,3 +51,5 @@ CONFIG_LOG_BACKEND_UART=n
 
 CONFIG_ASSERT=y
 CONFIG_HW_STACK_PROTECTION=y
+
+CONFIG_BOOTLOADER_MCUBOOT=y
```
```suggestion
```

hakonfam commented on samples/nrf5340/netboot/README.rst:
```diff
@@ -0,0 +1,73 @@
+nRF5340 Network core bootloader
+################################
+
+This bootloader sample implements an immutable first stage bootloader that has the capability to download update the application firmware on the network core of the nRF5340 SoC.
```
```suggestion
This bootloader sample implements an immutable first stage bootloader that has the capability to update the application firmware on the network core of the nRF5340 SoC.
```

hakonfam commented on samples/nrf5340/netboot/README.rst:
```diff
@@ -0,0 +1,73 @@
+     If the application core has configured a pending update, copy this to the application partition.
+     Once the copy is done, compare the SHA of the data in the application partition against the SHA provisioned in the firmware hex.
+
+#. Lock the flash of the application.
```
```suggestion
#. Lock the flash of the application partition.
```

hakonfam commented on subsys/pcd/src/pcd.c:
```diff
@@ -0,0 +1,68 @@
+	/* Signal complete by setting magic to DONE */
+	rsp->magic = PCD_CMD_MAGIC_DONE;
+	/* Wait for being rebooted */
+	while(1);
+	CODE_UNREACHABLE;
+	return 0;
```
This should be outside this function

sigvartmh replied:
Fixed.

hakonfam commented on include/dfu/pcd.h:
```diff
@@ -0,0 +1,90 @@
+	size_t offset;    /* Offset to store the flash image in */
+} __attribute__ ((aligned(4)));
+
+#define APP_CORE_SRAM_BASE_ADDRESS 0x20000000
```
DT?

sigvartmh committed:
fixups

hakonfam committed:
Botnet rebase (#32)

* manifest: nrfxlib to use new SoftDevice Controller

Updates the compatibility matrix and removes deprecated Kconfig
options.
See CHANGELOG for more details.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

* nfc: Fix NFC TNEP Poller CS Pin config

This commit fixes SPI CS Pin configuration for
the ST25R3911B in the TNEP Poller sample.

NCSDK-4934

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

* nfc: TNEP encode NDEF message directly in app data function

This commit provides encoding the NDEF messages directly in
the app_data function. This allows to encode locally definied
NDEF rocords.

NCSDK-4934

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

* nfc: Add callback for encoding the Initial NDEF Message

This commit provides the callback for encoding the NFC TNEP
Initial NDEF Message. This callback is called every time when
the TNEP Tag library has to encode the Initial Message.
This ensure that the Initial NDEF Records can be defined
locally in callback scope.

NCSDK-4934

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

* nfc: TNEP Tag Services in the ROM section

TNEP Tag Services are now keeped in ROM section
instead of passing them to the TNEP API.

NCSDK-4934

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

* nfc: Align TNEP documentation

This commits alignes the TNEP Tag documentation
to changes like encoding the TNEP Initial Messsage and
Application data

NCSDK-4934

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>
Signed-off-by: Francesco Servidio <francesco.servidio@nordicsemi.no>

* applications: asset_tracker: Add lte_lc event handler

Added event handler for lte_lc events in asset tracker.

Jira:CIA-53

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

* cmake: disassociate board from domain

Allow out of tree users with different boards to use domain
functionality without having patches.

Ref: NCSDK-5240

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* applications: asset_tracker: cloud_codec: Initialize struct

Initialize cloud_channel_data structs before use.

Jira:CIA-49

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

* applications: nrf_desktop: Rename missed config option

This is follow up of 781d4814ac91b529bf5cef80e150baa87c8d5c2a

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

* applications: nrf_desktop: Fix invalid reference to sub

Jira:DESK-1054

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

* dump WITH PCD

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* dump

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* dump

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* do we want this?

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* nrf5340_ netboot: Change how PCD_CMD address work and netboot sequence

Changed how `PCD_CMD_ADDRESS` is calculated. Removed unused buffer in
structure. Changed a bit how netboot boot sequence works.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

* west manifest

* Change cmd address

* Add RAM protection for copy instructionsw

* fix net core hex dep (#30)

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* fix dependency issue where net dfu was not generated (#31)

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

* fixups

Co-authored-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>
Co-authored-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>
Co-authored-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>
Co-authored-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>
Co-authored-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

hakonfam committed:
fix domains

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
